server/gitrest/src/externalStorageManager.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import Axios from "axios";
+import { config } from "process";
+import * as winston from "winston";
+
+export interface IExternalStorageManager {
+    read(tenantId: string, documentId: string): Promise<boolean>;
+
+    write(tenantId: string, ref: string, sha: string, update: boolean): Promise<void>;
+}
+
+/**
+ * Manages api calls to external storage
+ */
+export class ExternalStorageManager implements IExternalStorageManager {
+    constructor(private endpoint: string) {
+    }
+
+    public async read(tenantId: string, documentId: string): Promise<boolean> {
+        if (config.get("externalStorage:enabled") != "true") {
+            winston.info("External storage is not enabled");
+            return false;
+        }
+        await Axios.post<void>(
+            `${this.endpoint}/file/${tenantId}/${documentId}`,
+            {
+                headers: {
+                    "Accept": "application/json",
+                    "Content-Type": "application/json",
+                },
+            }).catch((error) => {
+                winston.error(`Axios error ${error} tenantId ${tenantId}`);
```

Comment:
Attaching the `messageMetaData` property to a log's data can be helpful for telemetry (see [Alfred's Lambda logging of tenantId and documentId](https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas/src/alfred/index.ts#L253-L258)).

Also, could this error message be made more readable? Something like 
```typescript
const messageMetaData = { tenantId, documentId };
winston.error(`Failed to read document: ${safeStringify(err, undefined, 2)}`, { messageMetaData });
```

Reply:
Currently we are throwing the error back to the caller, and catching it there - do you mean instead of that, just return false, and the caller will reject the promise?